kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/ColumnMapping.java

```diff
@@ -93,6 +100,68 @@ public static StructType convertToPhysicalSchema(
         }
     }
 
+    /** Returns the physical name for a given {@link StructField} */
+    public static String getPhysicalName(StructField field) {
+        if (field.getMetadata().contains(COLUMN_MAPPING_PHYSICAL_NAME_KEY)) {
+            return (String) field
+                    .getMetadata()
+                    .get(COLUMN_MAPPING_PHYSICAL_NAME_KEY);
+        } else {
+            return field.getName();
+        }
+    }
+
+    public static void verifyColumnMappingChange(
+            Map<String, String> oldConfig,
+            Map<String, String> newConfig) {
+        verifyColumnMappingChange(oldConfig, newConfig, true /* isNewTable */);
+    }
+
+    public static void verifyColumnMappingChange(
+            Map<String, String> oldConfig,
+            Map<String, String> newConfig,
+            boolean isNewTable) {
+        String oldMappingMode = getColumnMappingMode(oldConfig);
+        String newMappingMode = getColumnMappingMode(newConfig);
+
+        Preconditions.checkArgument(isNewTable ||
+                        allowMappingModeChange(oldMappingMode, newMappingMode),
+                "Changing column mapping mode from '%s' to '%s' is not supported",
+                oldMappingMode, newMappingMode);
+    }
+
+    public static boolean isColumnMappingModeEnabled(String columnMappingMode) {
+        return NAME_OR_ID.contains(columnMappingMode);
+    }
+
+    public static Metadata updateColumnMappingMetadata(
+            Metadata metadata,
+            String columnMappingMode,
+            boolean isNewTable) {
+        switch (columnMappingMode) {
+            case COLUMN_MAPPING_MODE_NONE:
+                return metadata;
+            case COLUMN_MAPPING_MODE_ID: // fall through
+            case COLUMN_MAPPING_MODE_NAME:
+                return assignColumnIdAndPhysicalName(metadata, isNewTable);
+            default:
+                throw new UnsupportedOperationException(
+                        "Unsupported column mapping mode: " + columnMappingMode);
+        }
+    }
+
+    ////////////////////////////
+    // Private Helper Methods //
+    ////////////////////////////
+
+    /** Visible for testing */
+    static int findMaxColumnId(StructType schema) {
```

Comment:
this doesn't handle nested schemas, right? are you planning on adding that in a subsequent PR?

I worry that adding it in a subsequent PR will leave us open to forgetting spots to add it ...

Reply:
+1.

Reply:
yeah I initially discussed this with @raveeram-db to add nested schema support in a separate PR

Reply:
I've opened https://github.com/delta-io/delta/issues/3427 to handled nested schema